contracts/minter-imago/src/contract.rs

```diff
@@ -0,0 +1,741 @@
+            }
+            mintable_tokens[0]
+        }
+    };
+
+    // Create mint msgs
+    let mint_msg = Cw721ExecuteMsg::Mint(MintMsg::<Empty> {
+        token_id: mintable_token_id.to_string(),
+        owner: recipient_addr.to_string(),
+        token_uri: Some(format!("{}/{}", config.base_token_uri, mintable_token_id)),
+        extension: Empty {},
+    });
+    let msg = CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: sg721_address.to_string(),
+        msg: to_binary(&mint_msg)?,
+        funds: vec![],
+    });
+    msgs.append(&mut vec![msg]);
+
+    // Remove mintable token id from map
+    MINTABLE_TOKEN_IDS.remove(deps.storage, mintable_token_id);
+    let mintable_num_tokens = MINTABLE_NUM_TOKENS.load(deps.storage)?;
+    // Decrement mintable num tokens
+    MINTABLE_NUM_TOKENS.save(deps.storage, &(mintable_num_tokens - 1))?;
+    // Save the new mint count for the sender's address
+    let new_mint_count = mint_count(deps.as_ref(), &info)? + 1;
+    MINTER_ADDRS.save(deps.storage, info.clone().sender, &new_mint_count)?;
+
+    // only allow one mint per address per block
+    let last_block_minter_sender = last_block_minted(deps.as_ref(), &info)?;
+    if last_block_minter_sender == env.block.height {
+        return Err(ContractError::MaxOneMintPerBlock {});
+    }
+    MINTER_LAST_BLOCK.save(deps.storage, info.clone().sender, &env.block.height)?;
+
+    Ok(Response::default()
+        .add_attribute("action", action)
+        .add_attribute("sender", info.sender)
+        .add_attribute("recipient", recipient_addr)
+        .add_attribute("token_id", mintable_token_id.to_string())
+        .add_attribute("network_fee", network_fee)
+        .add_attribute("pw_fee", pw_fee)
+        .add_attribute("mint_price", mint_price.amount)
+        .add_messages(msgs))
+}
+
+fn pw_fee_msg(
+    info: &MessageInfo,
+    fee: u128,
+    developer: Addr,
+) -> Result<Vec<CosmosMsg<StargazeMsgWrapper>>, FeeError> {
+    let payment = must_pay(info, NATIVE_DENOM)?;
+    if payment.u128() < fee {
+        return Err(FeeError::InsufficientFee(fee, payment.u128()));
+    };
+    let mut msgs: Vec<CosmosMsg<StargazeMsgWrapper>> = vec![];
+    let msg = BankMsg::Send {
+        to_address: developer.to_string(),
+        amount: coins(fee, NATIVE_DENOM),
+    };
+    msgs.push(CosmosMsg::Bank(msg));
+    (
+        Decimal::percent(PW_MINT_FEE_PERCENT),
+        fee,
+    );
+    Ok(msgs)
+}
+
+pub fn execute_burn_remaining(
+    deps: DepsMut,
+    _: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    // Check only admin
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized(
+            "Sender is not an admin".to_owned(),
+        ));
+    }
+    MINTABLE_NUM_TOKENS.save(deps.storage, &0)?;
+
+    let mut token_ids = vec![];
+    for mapping in MINTABLE_TOKEN_IDS.range(deps.storage, None, None, Order::Ascending) {
```

Comment:
This loops over all remaining tokens and burns them. Similar storage intensive calls require a fee. Does this call need a fee?

Reply:
great question, I think maybe just a small single fee it will be mostly a just 1 thing operation.

Reply:
Yes, this is just a one time operation.